VideoLocker/src/main/java/org/edx/mobile/util/NetworkUtil.java

```diff
@@ -168,4 +171,29 @@ public static boolean isSocialFeatureFlagEnabled(Context context, Config config)
         return isSocialEnabled && (NetworkUtil.isConnectedWifi(context) || !NetworkUtil.isOnSocialDisabledNetwork(context, config));
 
     }
+
+    /**
+     * Checks if a user has a working wifi connection or has allowed download over and active
+     * network connection, to allow a download request
+     *
+     * @param activity delegate of type {@link BaseFragmentActivity} to show proper error messages
+     * @return true if wifi is working or a user has allowed download over an active network
+     * connection, otherwise false
+     */
+
+    public static boolean isDownloadPossible(BaseFragmentActivity activity) {
```

Comment:
This should probably be renamed to `verifyDownloadPossible`, as it does perform error message display as well.